src/pages/resultsView/download/DownloadUtils.ts

```diff
@@ -22,20 +21,54 @@ export interface IDownloadFileRow {
 }
 
 export function generateOqlData(datum: GeneticTrackDatum,
-                                geneAlterationDataByGene?: {[gene: string]: IGeneAlteration}): IOqlData
+                                geneAlterationDataByGene?: {[gene: string]: IGeneAlteration},
+                                molecularProfileIdToMolecularProfile?: {[molecularProfileId:string]:MolecularProfile}): IOqlData
 {
     const proteinChanges: string[] = [];
     const fusions: string[] = [];
     const cnaAlterations: ISubAlteration[] = [];
     const proteinLevels: ISubAlteration[] = [];
     const mrnaExpressions: ISubAlteration[] = [];
-
+    const alterationTypes: string[] = [];
+    let isMutationProfiled = false;
+    let isFusionProfiled = false;
+    let isCnaProfiled = false;
+    let isMrnaExpProfiled = false;
+    let isProteinLevelProfiled = false;
+
+    //record the profile information
+    if (datum.profiled_in)
+    {
+        for (const profile of datum.profiled_in)
```

Comment:
This logic is assuming that if it is not in the `profiled_in` list that it is not profiled. But that may not always be the case. I would instead go over the `not_profiled_in` array to find out the not profiled ones. So maybe also rename your flags to `isMutationNotProfiled`, `isCnaNotProfiled`, etc.

Reply:
@onursumer Thanks Onur, I just updated my commit. Please take a look, appreciated.

Reply:
@dippindots Thanks for the changes, I just found one more thing, can you also update that one?